src/main/java/service/PatientImpl.java

```diff
@@ -37,7 +37,7 @@ public synchronized boolean bookAppointment(
     Optional<Appointment> appointment = database.findByTypeAndId(type, appointmentId);
     ArrayList<PatientId> patientIds = appointment.orElseThrow().addPatient(patientId);
     int capacity = appointment.get().getRemainingCapacity();
-    database.update(new Appointment(appointmentId, capacity, patientIds), appointmentId);
+    database.update(new Appointment(appointmentId, capacity, patientIds), type);
```

Comment:
No need to change capacity. `addPatient()` will do the job.
Just update the appointment after `addPatient` like
```java
database.update(appointment, type);
```